light-client/src/commitment.rs

```diff
@@ -52,7 +52,9 @@ pub fn verify_proof(
 ) -> Result<(), Error> {
     let value = verify(root, proof, key)?;
 
-    let expected_value = expected_value.as_ref().map(|e| rlp::encode(e).to_vec());
+    let expected_value = expected_value
+        .as_ref()
+        .map(|e| rlp::encode(&trim_left_zero(e).to_vec()).to_vec());
     if value != expected_value {
         return Err(Error::UnexpectedStateValue(
```

Comment:
Can you add `value` to `UnexpectedStateValue` error? It would be helpful for debug.

Reply:
done